recipes/gf-complete/all/conanfile.py

```diff
@@ -0,0 +1,110 @@
+    name = "gf-complete"
+    description = "A library for Galois Field arithmetic"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/ceph/gf-complete"
+    license = "BSD-3-Clause"
+    topics = ("galois field", "math", "algorithms")
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "neon": [True, False],
+        "sse": [True, False],
+        "avx": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "neon": True,
+        "sse": True,
+        "avx": False
+    }
+
+    _source_subfolder = "source_subfolder"
+    _autotools = None
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-" + self.version, self._source_subfolder)
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        if tools.os_info.is_windows and not self.settings.compiler == "gcc":
+            # Building on Windows is currently only supported using the MSYS2
+            # subsystem. In theory, the gf-complete library can be build using
+            # MSVC. However, some adjustments to the build-system are needed
+            # and the CLI tools cannot be build.
+            #
+            # A suitable profile for MSYS2 can be found in the documentation:
+            # https://github.com/conan-io/docs/blob/b712aa7c0dc99607c46c57585787ced2ae66ac33/systems_cross_building/windows_subsystems.rst
+            raise ConanInvalidConfiguration(
+                "Windows is only supported using the MSYS2 subsystem")
+
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+
+        self._autotools = AutoToolsBuildEnvironment(
+            self, win_bash=tools.os_info.is_windows)
+
+        with tools.environment_append(self._autotools.vars):
+            # run autoreconf
+            self.run("./autogen.sh", win_bash=tools.os_info.is_windows)
+
+        if "x86" in self.settings.arch:
+            self._autotools.flags.append('-mstackrealign')
+
+        if "fPIC" in self.options:
+            self._autotools.fpic = self.options.fPIC
+
+        configure_args = [
+            "--enable-shared=%s" % ("yes" if self.options.shared else "no"),
+            "--enable-static=%s" % ("no" if self.options.shared else "yes")
+        ]
+
+        # enabled by default
+        if not self.options.neon:
+            configure_args.append("--disable-neon")
+        if not self.options.sse:
+            configure_args.append("--disable-sse")
+
+        # disabled by default
+        if self.options.avx:
+            configure_args.append("--enable-avx")
+
+        self._autotools.configure(args=configure_args)
```

Comment:
```suggestion
        self._autotools.configure(args=configure_args, configure_dir=self._source_subfoler)
```

You don't need to use chdir for build and package

Reply:
I'll have to chdir in order to run the autogen.sh script instead, but at least it's confined in the `_configure_autotools` method this way.